### PR TITLE
feature/TASK-3_json-storage

### PR DESCRIPTION
**Pull request by JeriRov**

### CODE REVIEW

File: 03_json_storage/src/services/json.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    private dataDirectory: string = path.join(__dirname, '../..', 'json_data');
+
+    constructor() {
+        if (!fs.existsSync(this.dataDirectory)) {
+            fs.mkdirSync(this.dataDirectory);
+        }
+    }
+
+    storeData(jsonPath: string, data: Json) {
+        if (!jsonPath) throw new Error('jsonPath is required');
```
Comment (milkanuch) [inline 03_json_storage/src/services/json.ts]: I don't like one lines if((

File: 03_json_storage/src/models/json.types.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export type Data = string | number | boolean | null | Data[];
+export type Json = Record<string, Data> | Data[];
```
Comment (milkanuch) [inline 03_json_storage/src/models/json.types.ts]: I think this naming is bad,  don't name type like JSON or Data, it's really general names.

Comment (JeriRov) [reply 03_json_storage/src/models/json.types.ts]: You're right, so I renamed those types like this:
```typescript
export type JsonDataValue = string | number | boolean | null | JsonDataValue[];
export type JsonObject = Record<string, JsonDataValue> | JsonDataValue[];
```

File: 03_json_storage/src/services/json.ts (hunk 1 above)
Comment (JeriRov) [reply 03_json_storage/src/services/json.ts]: Well, I changed that!

**Pull request merged**